packages/shared/lib/sdk/sync.ts

```diff
@@ -570,7 +571,14 @@ export class NangoSync extends NangoAction {
                     lastSyncDate: date
                 }
             });
-            return response.status <= 299;
+            if (response.status > 299) {
+                console.log(
+                    `Request to persist API (setLastSyncDate) failed: errorCode=${response.status} response='${JSON.stringify(response.data)}'`,
+                    JSON.stringify(this)
+                );
+                throw new Error(`cannot set lastSyncDate for sync '${this.syncId}'`);
```

Comment:
operation results are rarely (or even ever) checked in integration. Let's throw an error to make it fails loudly instead

Reply:
How does this error surface to the end user?

Reply:
if it is not catched and handled by the integration, it is captured as a failure of the sync
<img width="1249" alt="Screenshot 2024-02-01 at 07 57 11" src="https://github.com/NangoHQ/nango/assets/233326/8873d00c-4303-41e6-8787-e0024ad85309">